Add general polder deployment

pantierra opened the pull request:
This PR introduces a general `polder` chart that has as dependencies the `webapp-polder` and `eoapi`. It also fixes the failing tests of the CI pipeline.

Follow-up to #15 

Closes #13

pantierra committed:
Added general polder deployment with its dependencies.

ciaransweet commented on deployment/helm/.gitignore:
```diff
@@ -0,0 +1,37 @@
+# General files for the project
```
Do we need another `.gitignore` if we already have a project level one? Can this be merged to the top level one?

*edit* I'm aware this is autogen'd by helm

pantierra replied:
You are right. I also dislike having several `.gitignore` files. I just merged it with the root level `.gitignore`

ciaransweet commented on deployment/helm/polder/values.yaml:
```diff
@@ -0,0 +1,15 @@
+fullnameOverride: "test-polder"
+
+# polder custom configuration
+mapboxToken: "pk.eyJ0IjoiZGV2c2VlZCIsImEiOiJnUi1mbkVvIn0.018aLhX0Mb0tdtaT2QNe2Q"
```
Whatever this token is. We should definitely remove it *and* regenerate it on Mapbox... Given this repo is public, I assume this is now exposed and vulnerable

pantierra replied:
Oh, I am sorry. I was assuming this was just to be some dummy token.

Shall we handle normal k8s secrets instead? And share the file in some private repo.

emmanuelmathot commented on deployment/helm/polder/values.yaml:
```diff
@@ -0,0 +1,15 @@
+
+# polder custom configuration
+mapboxToken: ""
+stacApi: "https://rx2hna9pbg.execute-api.eu-central-1.amazonaws.com"
```
Those are links for WHIS, I would replace them with the default URL generated by the default helm deploy on kubernetes. (e.g. polder-eopi-stac)

pantierra replied:
Done.

emmanuelmathot committed:
make charts check work

emmanuelmathot committed:
Full helm CI/CD testing
- Update Helm chart and workflow for webapp container build
- Added postgres operator install
- chart-testing enabled with kind installation

emmanuelmathot commented:
I updated ther workflow to make the k8s integration test working but there are still issue when deploying eoapi-k8s

emmanuelmathot commented:
There is [something wrong with the vector API in eoAPI](https://github.com/developmentseed/polder/actions/runs/11701112077/job/32586512382?pr=16#step:14:4451) that is not captured by the eoAPI helm chart and thus gives a false positive. A readiness probe probably needs to be set up correctly.

pantierra committed:
Adjusted internal urls.

pantierra commented:
Many thanks for the review! I adjusted the urls, and created a follow up ticket for the additional tests #17.
